src/Common/tests/System/Collections/IEnumerable.Generic.Tests.cs

```diff
@@ -283,16 +276,14 @@ protected enum EnumerableOrder
                         "enumerator.MoveNext() returned true past the expected end.");
                 }
 
-                if (!Enumerator_Current_UndefinedOperation_Throws)
+                if (Enumerator_Current_UndefinedOperation_Throws)
```

Comment:
Old code did skip testing when `Enumerator_Current_UndefinedOperation_Throws` equal false.
When  `Enumerator_Current_UndefinedOperation_Throws` equal false we should test that code does not throw exception `var cur = enumerator.Current;`.